usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -6882,10 +6882,10 @@ l2arc_write_buffers(spa_t *spa, l2arc_dev_t *dev, uint64_t target_sz)
 
 	ASSERT3U(write_asize, <=, target_sz);
 	ARCSTAT_BUMP(arcstat_l2_writes_sent);
-	ARCSTAT_INCR(arcstat_l2_write_bytes, write_asize);
+	ARCSTAT_INCR(arcstat_l2_write_bytes, write_psize);
 	ARCSTAT_INCR(arcstat_l2_size, write_sz);
```

Comment:
Optional: to make things clearer it would be nice if we renamed `arcstat_l2_size` to `arcstat_l2_lsize`. Also rename `write_sz` to `write_lsize`. This would clarify that we're tracking both the logical size and physical size of I/Os that have been done to the l2arc. I note this as optional since it's not really part of your original fix so if you want to drop this then I can make that change.